internal/ingress/annotations/authreq/main.go

```diff
@@ -193,6 +215,29 @@ func (a authReq) Parse(ing *networking.Ingress) (interface{}, error) {
 		klog.V(3).InfoS("auth-cache-key annotation is undefined and will not be set")
 	}
 
+	keepaliveConnections, err := parser.GetIntAnnotation("auth-keepalive", ing)
+	if err != nil {
+		klog.V(3).InfoS("auth-keepalive annotation is undefined and will be set to its default value")
+		keepaliveConnections = defaultKeepaliveConnections
+	}
+	// NOTE: upstream block cannot reference a variable in the server directive
+	if keepaliveConnections != 0 && strings.IndexByte(authURL.Host, '$') != -1 {
+		klog.V(3).InfoS("auth-url annotation contains $ in the host:port part, setting auth-keepalive to 0")
```

Comment:
maybe add as a warning :)

In a future, @strongjz is working on generating events. Maybe I should discuss with @thockin if there's some value on adding conditions in ingress status as well